bezpy/mt/site.py

```diff
@@ -243,8 +243,19 @@ def download_waveforms(self):
 
         # Magnetic Field
         self.waveforms[["FE", "FN", "FZ"]] *= 0.01  # nT
-        # Electric Field
-        self.waveforms[["QN", "QE"]] *= 2.44141221047903e-05  # mV/km
+        # Electric voltage
+        self.waveforms[["QN", "QE"]] *= 2.44141221047903e-06  # mV
+        # Electric Field (set default length of dipole as 100m)
+        self.waveforms[["QN", "QE"]] *= (1000.0/100.0)  # mV/km
```

Comment:
Would it be possible to do this conversion after all of your new length calculations? i.e. move this after the next try/except block. Or, put the default length into the except block.

```suggestion

# Get the run times, as dipoles could be different lengths for different runs
mask = ((self.waveforms.index > self.runinfo[runid]['Start']) &
               (self.waveforms.index < self.runinfo[runid]['End']))
try:
    dipole_x = self.runinfo[runid]['Ex']
except:
    # Default dipole length = 100 m
    dipole_x_length = 100.

try:
    dipole_y = self.runinfo[runid]['Ey']
except:
    # Default dipole length = 100 m
    dipole_x_length = 100.

self.waveforms["QN"].loc[mask] *= (1000.0/dipole_length_x) # mV/km
self.waveforms["QE"].loc[mask] *= (1000.0/dipole_length_y) # mV/km
```

I'm not entirely sure that will work, but I think that logic is easier to follow perhaps. Give that a thought.

Reply:
That's much easier to understand. Mask has to be under `try` because we don't know starting and ending time if XML file does not have run information.